core/src/main/java/org/testcontainers/containers/image/pull/policy/AgeBasedPullPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.testcontainers.containers.image.pull.policy;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.image.ImageData;
+
+/**
+ * An ImagePullPolicy which pulls the image from a remote repository only if its created date is older than maxAge
+ */
+@Slf4j
+public class AgeBasedPullPolicy implements ImagePullPolicy {
+
+    private long maxAge;
+    private TimeUnit unit;
+
+    /**
+     * @param maxAge - Maximum age of image (based on image Created parameter)
+     * @param unit - The TimeUnit to use (MINUTES, HOURS, DAYS, etc.)
+     */
+    public AgeBasedPullPolicy(long maxAge, TimeUnit unit) {
+        this.maxAge = maxAge;
+        this.unit = unit;
+    }
+
+    @Override
+    public boolean shouldPull(ImageData image) {
+        boolean result = unit.convert(Instant.now().getEpochSecond() - image.getCreated(), TimeUnit.SECONDS) > maxAge;
+        if (result) {
+            log.trace("Should pull image with tags: {}", Arrays.asList(image.getRepoTags()));
```

Comment:
please log what happens, not what should happen. Something like:
```suggestion
            log.trace("Pulling an expired image with tags: {}. Age '{}' is higher than max age '{}'", Arrays.asList(image.getRepoTags()), age, maxAge);
```

Reply:
The pulling doesn't actually happen there, wouldn't it be confusing?